**# **Exploración y reconocimiento de los datos de la tabla Store****

# Tabla de Contenido
- [TABLA Walmart_Sales](#tabla-Walmart_Sales)
  - [Columna Store](#columna-Store)
  - [Columna Date](#Columna-date)
  - [Columna Weekly_Sales](#Columna-weekly_sales)
  - [Columna Holiday_Flag](#Columna-Holiday_Flag)
  - [Columna Temperature](#Columna-temperature)
  - [Columna Fuel_Price](#Columna-fuel_price)
  - [Columna CPI](#Columna-cpi)
  - [Columna Unemployment](#Columna-unemployment)
  - [Resultados y conclusiones](#Resultados-y-conclusiones)

### **TABLA Walmart_Sales**
<span style="font-size: 18px;">
La exploración de los datos de la tabla  "Walmart_Sales" consta de los siguientes pasos:

1. Consulta de la cantidad de registros en la tabla.
1. Visualización de la tabla.
1. Visualización de registros NULL en la columna Store.
1. Visualización de registros NULL en la columna Date.
1. Visualización de registros NULL en la columna Weekly_Sales.
1. Visualización de registros NULL en la columna Holiday_Flag.
1. Visualización de registros NULL en la columna Temperature.
1. Visualización de registros NULL en la columna Fuel_Price. 
1. Visualización de registros NULL en la columna CPI.
1. Visualización de registros NULL en la columna Unemployment.
</span>

In [1]:
-- # 1.
USE TFM;
SELECT COUNT(*) as cantidad_registro from Walmart_sales;

(1 row affected)

Total execution time: 00:00:00.006

cantidad_registro 6435

<span style="font-size: 18px;">
Se puede observar que esta tabla tiene un total de 6435 registros. Ahora veamos algunos datos de la tabla.
</span>

In [2]:
-- # 2.
select TOP 10 *from Walmart_sales;

(10 rows affected)

Total execution time: 00:00:00.004

Store Date Weekly_Sales Holiday_Flag Temperature Fuel_Price CPI Unemployment 1 2010-02-05 16436909 0 42.31 2.572 2110963584 8.106 1 2010-02-12 164195744 1 38.51 2.548 2112421760 8.106 1 2010-02-19 161196816 0 39.93 2.514 2112891392 8.106 1 2010-02-26 140972752 0 46.63 2.561 2113196416 8.106 1 2010-03-05 155480672 0 46.5 2.625 2113501440 8.106 1 2010-03-12 143954160 0 57.79 2.667 2113806464 8.106 1 2010-03-19 147251584 0 54.58 2.72 211215632 8.106 1 2010-03-26 140442992 0 51.45 2.732 2110180480 8.106 1 2010-04-02 159496832 0 62.27 2.719 2108204544 7.808 1 2010-04-09 154541856 0 65.86 2.77 2106228608 7.808

<span style="font-size: 18px;">
En la tabla se pueden observar las siguientes características:

1. Columna "Store": Contiene el ID único de cada tienda.
1. Columna "Date": Los datos están en formato de fecha YYYY-MM-DD.
1. Columna "Weekly_Sales": Incluye valores numéricos enteros.
1. Columna "Holiday_Flag": Presenta valores numéricos enteros.
1. Columna "Temperature": Contiene valores numéricos con decimales.
1. Columna "Fuel_Price": Incluye valores numéricos con decimales.
1. Columna "CPI": Contiene valores numéricos enteros.
1. Columna "Unemployment": Presenta valores numéricos con decimales.


A continuación, se analizarán estas observaciones con mayor detalle.
</span>

## **Columna Store**
<span style="font-size: 18px;">
En esta columna nos enfocamos en :

- <b>Búsqueda de valores null. </b> 
- <b>Búsqueda de datos duplicados: </b> Específicamente, buscamos combinaciones repetidas de tienda (Store) y fecha (Date) .


</span>

In [3]:
-- # 3
select * from Walmart_Sales
where Store is NULL 
or Store LIKE '%[^0-9]%';


(0 rows affected)

Total execution time: 00:00:00.018

Store Date Weekly_Sales Holiday_Flag Temperature Fuel_Price CPI Unemployment

In [4]:
SELECT Store, [Date], count(*)
FROM Walmart_Sales
GROUP BY Store, [Date]
HAVING count(*) > 1;


(0 rows affected)

Total execution time: 00:00:00.014

Store Date (No column name)

<span style="font-size: 18px;">Podemos ver que en la columna Store no hay valores faltantes, todos los datos son numéricos y no existen datos duplicados en la combinación Store y Date.</span>

## **Columna Date**

<span style="font-size: 18px;">
En esta columna nos enfocamos en:

- <b>Búsqueda de valores null</b>.
</span>

In [5]:
-- # 4
SELECT * 
FROM Walmart_Sales
WHERE [Date] IS NULL;

(0 rows affected)

Total execution time: 00:00:00.002

Store Date Weekly_Sales Holiday_Flag Temperature Fuel_Price CPI Unemployment

<span style="font-size: 18px;">
Podemos ver que en la columna Date no hay valores faltantes y todos los datos son fechas.
</span>

## **Columna Weekly_Sales**
<span style="font-size: 18px;">
En esta columna nos enfocamos en:

- <b>Búsqueda de valores null. </b> 

In [6]:
-- # 5.
select *from Walmart_Sales
where Weekly_Sales is NULL

(0 rows affected)

Total execution time: 00:00:00.003

Store Date Weekly_Sales Holiday_Flag Temperature Fuel_Price CPI Unemployment

<span style="font-size: 18px;">
Podemos ver que en la columna Walmart_Sales no hay valores faltantes.
</span>

## **Columna Holiday_Flag**
<span style="font-size: 18px;">
En esta columna nos enfocamos en:

- <b>Búsqueda de valores null. </b> 
- <b>Búsqueda de valores no numéricos.</b>

In [7]:
-- # 6.
select *from Walmart_Sales
where Holiday_Flag is NULL
or Holiday_Flag LIKE '%[^0-9]%';

(0 rows affected)

Total execution time: 00:00:00.017

Store Date Weekly_Sales Holiday_Flag Temperature Fuel_Price CPI Unemployment

<span style="font-size: 18px;">

Esta columna tampoco tiene valores faltantes.

</span>

## **Columna Temperature**
<span style="font-size: 18px;">
En esta columna nos enfocamos en:

- <b>Búsqueda de valores null. </b> 

In [8]:
-- # 7. 
select *from Walmart_Sales
where Temperature is NULL

(0 rows affected)

Total execution time: 00:00:00.002

Store Date Weekly_Sales Holiday_Flag Temperature Fuel_Price CPI Unemployment

<span style="font-size: 18px;">

En la columna Temperatura tampoco hay valores faltantes.

</span>

## **Columna Fuel_Price**

<span style="font-size: 18px;">

En esta columna nos enfocamos en:

- <b>Búsqueda de valores null. </b>

In [9]:
-- # 8.
select *from Walmart_Sales
where Fuel_Price is NULL;

(0 rows affected)

Total execution time: 00:00:00.002

Store Date Weekly_Sales Holiday_Flag Temperature Fuel_Price CPI Unemployment

<span style="font-size: 18px;">

En la columna Fuel_Price tampoco hay valores faltantes.

</span>

## **Columna CPI**

<span style="font-size: 18px;">

En esta columna nos enfocamos en:

- <b>Búsqueda de valores null. </b>

In [10]:
-- # 9.
select *from Walmart_Sales
where CPI is NULL;

(0 rows affected)

Total execution time: 00:00:00.003

Store Date Weekly_Sales Holiday_Flag Temperature Fuel_Price CPI Unemployment

<span style="font-size: 18px;">

En la columna CPI tampoco hay valores faltantes.

</span>

## **Columna Unemployment**

<span style="font-size: 18px;">

En esta columna nos enfocamos en:

- <b>Búsqueda de valores null. </b>

In [11]:
-- # 10.
select *from Walmart_Sales
where Unemployment is NULL;

(0 rows affected)

Total execution time: 00:00:00.002

Store Date Weekly_Sales Holiday_Flag Temperature Fuel_Price CPI Unemployment

## **Resultados y conclusiones**

<span style="font-size: 18px;">
A modo de resumen tenemos los siguientes resultados:

1. Tras revisar la tabla, se observó que todas las columnas están libres de registros NULL, lo cual es un resultado positivo. Sin embargo, se identificó que es necesario transformar todas las columnas al formato adecuado. Actualmente, la única columna que tiene un formato válido y apropiado para el tipo de datos que almacena es la columna Date.
1. Al revisar la agrupación Store y Date, se observó que no hay registros con la misma tienda y la misma fecha en la tabla. Es decir, la tabla no contiene registros duplicados.


<b>Conclusiones:</b>

1. Es crucial revisar y ajustar los formatos de las demás columnas para garantizar que los datos se almacenen y se procesen correctamente. Esto no solo mejorará la integridad de los datos, sino que también facilitará futuras consultas y análisis.

</span>